domain/application/state/application.go

```diff
@@ -378,7 +378,7 @@ func (st *ApplicationState) AddUnits(ctx context.Context, applicationName string
 
 // GetUnitUUID returns the UUID for the named unit, returning an error
 // satisfying [applicationerrors.UnitNotFound] if the unit doesn't exist.
-func (st *ApplicationState) GetUnitUUID(ctx domain.AtomicContext, unitName string) (string, error) {
+func (st *ApplicationState) GetUnitUUID(ctx domain.AtomicContext, unitName string) (coreunit.UUID, error) {
```

Comment:
Is this a convention I was not aware of? On the machine domain `GetMachineUUID()` returns a string.

Reply:
I think we're trying to use specific types for all these UUIDs, I think the unit one is in the process of being wired up.

Reply:
yep, as Alastair said. We're moving over to that approach, but it's an on-going process